<launch>
  <!-- INPUT ARGUMENTS -->
  <!-- Mode for training spatial models this session: May be set to "new", "aggregate" or "off" -->
  <arg name="train_mode" default="new"/>
  <!-- Whether to save newest train data to corpus_file -->
  <arg name="save_train_data" default="false"/>
  <!-- Location in which to save/retrieve processed data -->
  <arg name="corpus_file" default="$(find hrc_discrim_learning)/train/spatial_corpus.csv"/>

  <!-- Whether to use the aruco stuff -->
  <arg name="use_aruco" default="false"/>
  <!-- File containing object locations, if not using aruco -->
  <arg name="obj_location_file" default="$(find hrc_discrim_learning)/train/object_locations.json"/>
  <!-- File containing object id-feature mappings -->
  <arg name="obj_mapping_file" default="$(find hrc_discrim_learning)/train/perceptual_mapping.json"/>
  <!-- File containing newest training (id, utterance) data -->
  <arg name="train_data_file" default="$(find hrc_discrim_learning)/train/spatial_train_data.json"/>
  <!-- Whether to save the model trained this session -->
  <arg name="save_model" default="false"/>
  <!-- If save_model = True, folder in which to save model -->
  <arg name="save_dest" default="$(find hrc_discrim_learning)/model/spatial"/>

  <!-- ROS PARAMETERS -->
  <param name="hrc_discrim_learning/use_aruco"          value="$(arg use_aruco)"/>
  <param name="hrc_discrim_learning/obj_location_file"  value="$(arg obj_location_file)"/>

  <param name="hrc_discrim_learning/obj_mapping_file"   value="$(arg obj_mapping_file)"/>

  <param name="hrc_discrim_learning/save_model"         value="$(arg save_model)"/>
  <param name="hrc_discrim_learning/save_dest"          value="$(arg save_dest)"/>
  <param name="hrc_discrim_learning/train_mode"         value="$(arg train_mode)"/>
  <param name="hrc_discrim_learning/train_data_file"    value="$(arg train_data_file)"/>
  <param name="hrc_discrim_learning/save_train_data"    value="$(arg save_train_data)"/>
  <param name="hrc_discrim_learning/corpus_file"        value="$(arg corpus_file)"/>


  <!-- LAUNCH -->
  <group if="$(arg use_aruco)">
    <include file="$(find hrc_discrim_learning)/launch/basic_aruco.launch"/>
  </group>

  <node pkg="hrc_discrim_learning" type="auto_train.py" name="scripted_training" respawn="false" output="screen"/>

  <node pkg="hrc_discrim_learning" type="trainer_spatial.py" name="train_spatial" respawn="false" output="screen"/>

</launch>
